front/src/app/todo-list/todo-list.component.ts

```diff
@@ -0,0 +1,43 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, OnInit } from '@angular/core';
+import { ApiService, Lists } from '../shared/services/Api.service';
+import { FormBuilder, FormControl, FormGroup } from '@angular/forms';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-todo-list',
+  templateUrl: './todo-list.component.html',
+  styleUrl: './todo-list.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class TodoListComponent implements OnInit{
+
+  allTaskLists: Lists;
+
+  radioBtnGroup: FormGroup = new FormGroup ({
+    activeTaskList: new FormControl()
+  })
+
+  constructor(public apiService: ApiService, private changeDetectorRef: ChangeDetectorRef, public formBuilder: FormBuilder, private route: ActivatedRoute) {
+    
+  }
+
+  ngOnInit(): void {
+    this.loadTaskLists();
+    this.route.queryParams
+      .subscribe(params => {
+        if (params['list'] != undefined) {
+          this.radioBtnGroup = new FormGroup ({
+            activeTaskList: new FormControl(params['list'])
+          })
+        }
+      })
+  }
+  
+  loadTaskLists() {
+    this.apiService.getLists()
+      .subscribe(data => {
```

Comment:
there's a potential memory leak here - read about unsubscribing https://blog.bitsrc.io/6-ways-to-unsubscribe-from-observables-in-angular-ab912819a78f

Reply:
Still a potential memory leak here if the user left the page before data is loaded.

You correctly fixed router subscription, but not this one